build.bat

```diff
@@ -78,9 +78,22 @@ if errorlevel 1 goto error
 goto end
 
 :build_boost
-set boost_build_flags=toolset=msvc-14.0 variant=release link=static threading=multi runtime-link=static
-set boost_libs=--with-date_time --with-filesystem --with-locale --with-regex --with-signals --with-system --with-thread
-cd %BOOST_ROOT%
+
+set boost_build_flags=toolset=msvc-%VisualStudioVersion%^

```

Comment:
Should not use %VisualStudioVersion% here.
The number is platform toolset, not Visual Studio version.
For example msvc-14.1 for Visual Studio 15 2017.

https://stackoverflow.com/questions/42730478/version-numbers-for-visual-studio-2017-boost-and-cmake/42738533#42738533